### ISSUE TITLE
[Question] How to customize specific  UIBarButton?

### ISSUE THREAD

**Issue by LinkedHK** [chiunam/CTAssetsPickerController#179]
Thank you for a helpful library.

I've spent a whole day figuring out how I can customize a specific UIBarButton.
The code in the demo app https://github.com/chiunam/CTAssetsPickerController/blob/master/CTAssetsPickerDemo/Examples/CTApperanceViewController.m#L84
doesn't give  a clear example of how it can be done
I want to change title of a button "Done" to "Upload" ., and also I would like to change the design  of buttons. Currently it displays 
On the left is shown current button design . and desired on the right.

![4-ui-issue2](https://cloud.githubusercontent.com/assets/2436637/11376725/4761ca9a-931d-11e5-92bf-bdef419f7dfa.png)

I've just looking into a source code. there is no way to access  ui items `UIBarButtonItem * cancelButton ` ,  `UIBarButtonItem * doneButton ` in  `CTAssetCollectionViewController`

Therefore I presume they can't be customized individually. moreover class `CTAssetCollectionViewController` cannot be extended  

Thank you!

**Comment by chiunam** [comment]
Hi @LinkedHK, really nice to know someone from Hong Kong!

Re the first question, you can override "Done" button title by assigning `doneButtonTitle` property in v3.2.0 or above.

Re the second one, I believe you have another global UIAppearance set so all styles of UIBarButton are overridden. You could set the style by using `appearanceWhenContainedIn:` appearance proxy and pass `[CTAssetsPickerController class]` as parameter. Please refer to the documentation of UIAppearance.

**Comment by LinkedHK** [comment]
Hi., @chiunam  I appreciate your for help.  However. It only solved part of problem.  According to design requirements (see at the right screenshot above)  The "done" button should be displayed only inside of the album or if user picks some images it may also be displayed outside of album. Since there is is no public method available to detect  if user is inside an album I had to look into the source code of your library; the class `CTAssetCollectionViewController` has the method `isTopViewController` which suits my need . In order to solve my problem I  created a subclass of classes CTAssetCollectionViewController and CTAssetsPickerController.. and achieved my goal using reflection. So Basically I hacked your Library :) 
Thanks again for the help!
P.S I'm not a Hong Konger.. and live 4000 km Away.(Lookup in google the Bishkek city) . but I love Hong Kong and I have a few friends here :)
